src/components/PageRefreshButton/PageRefreshButton.tsx

```diff
@@ -25,10 +32,43 @@ export const PageRefreshButton = () => {
     }
   };
 
+  const catchUserEye = (elem: HTMLButtonElement) => {
+    elem.scrollIntoView({ behavior: 'smooth' });
+    setCatchEye(true);
+    setIsRefreshing(true);
+    setTimeout(() => setIsRefreshing(false), 2000);
+    setTimeout(() => setCatchEye(false), 5000);
+  };
+
+  useEffect(() => {
+    const handleBeforeUnload = (event: BeforeUnloadEvent) => {
+      event.preventDefault(); // Recommended by MDN
+      event.returnValue = true; // Included for legacy support, e.g. Chrome/Edge < 119
+
+      onRefreshClick().then(() => {
+        if (ref && 'current' in ref && ref.current) {
```

Comment:
because ref is of type either some function that I don't know how to handle here or object with current property, therfore typescript won't allow me to simply use current